### DISCUSSION TITLE
Balkonkraftwerk als zweiter Erzeuger in eine PV Anlage integrieren

### DISCUSSION THREAD

**Discussion by Nulleinspeiser-mit-Batteriespeicher** [tbnobody/OpenDTU#1726] [Ideas]
Hallo zusammen, 
Nach ein paar Jahren in kleinem, kommt dieses Jahr eine große Anlage aufs Dach. 
Das BKW wird weiter betrieben und soll als weiterer Erzeuger in einen Fronius Wechselrichter und Web-Interface integriert werden, damit das Haus kein Strom produziert, sondern verbraucht.
Das ich neue Kabel durchs Haus und einen zweiten Messpunkt in den Zählerschrank einbauen kann ist mir bewusst, aber vielleicht gibt es hier eine Idee wie die DTU "zum zweiten Energiezähler" (mittels Modbus TCP?) umgebaut werden kann.
Ich kann es leider nicht selber programmieren, aber vielleicht hat oder hatte jemand das gleiche Problem und konnte eine Lösung dafür finden.
Viele Grüße

**Comment by Abraham1220** [comment]
Hallo, 

stehe vor der gleichen Aufgabenstellung.

Ich habe zwar beschlossen zwecks Genauigkeit einen eigenen Smartmeter von Fronius einzusetzen, weil ich im Endausbau fünf Mikrowechselrichter neben dem Fronius Gen24 haben werde, und mir das mit zwei OpenDTU's dann zu kompliziert wird.

D.h. modulweises Tracing via OpenDTU. Korrekte Darstellung im Fronius Solarweb und Gesamtertrag durch einen zweiten Smartmeter via Modbus TCP lesend am Fronius Gen24. Soweit die Theorie.

Aber ich habe bei meiner Rechereche folgenden OpenDTU Fork gefunden, der genau das macht, was Du willst via Modbus TCP. Fronius pollt jede Sekunde die OpenDTU: [https://github.com/AloisKlingler/OpenDTU-FroniusSM-MB](url)

Artikel dazu: [https://www.photovoltaikforum.com/thread/205384-hoymiles-mikrowechselrichter-mit-opendtu-in-solarweb-aufnehmen-ohne-smartmeter/](url)

Viel Glück. Berichte vielleicht hier über Deinen Erfolg.

LG,
Michael ...

**Comment by Nulleinspeiser-mit-Batteriespeicher** [comment]
Hallo Michael,

Besten Dank für deinen Tip mit dem Fork von

https://github.com/AloisKlingler/OpenDTU-FroniusSM-MB

Nach einigem Suchen habe ich einen Erfahrungsaustausch gefunden:

https://www.photovoltaikforum.com/thread/205384-hoymiles-mikrowechselrichter-mit-opendtu-in-solarweb-aufnehmen-ohne-smartmeter/

Bei deiner Konstellation macht ein eigenes Smartmeter definitiv Sinn. Das hört sich für mich nach 5 Kw/p per Microwechselrichter an. Und so wie du das aufbauen möchtest sollte das auch klappen. Smartmeter für den Gesamtertrag im Fonuis und die OpenDTU für den Rest.

Bei mir möchte ich nur nicht, dass das Haus mit dem alten Balkonkraftwerk (Damals als Sichtschutz ins Balkongeländern eingebaut) in der Statistik Strom erzeugt und alle Werte und Auswertungen dann doch zerlegt. 

Soweit hört sich der Fork gut an, es werden keine doppelten Werte mehr übermittelt mit der Version 13.11.2023. Ein 10 bis 15 Sekunden Intervall, der dort empfohlen wird, würde mir schon reichen um die 300 bis 500 Watt die vom hoymiles produziert werden im Fronius aufzuzeichnen und die Statistik sauber zu halten.

Und wenn ich anfange die Anlage aufzubauen, vielleicht fällt mir noch ein Stromkabel mehr durchs Haus für eine eigene Leitung des Bkw und ein zweites Smartmeter. Das kann man ggf. dann für einen anderen Verbraucher verwenden wenn alles mit der open dut klappen sollte.

Ich danke dir für deine Hilfe, ich werde das im Sommer testen und meine Erfahrung mitteilen.

LG Raphael

**Reply by Abraham1220** [reply]
Hallo Raphael,

habe zerstreute Flächen und am Haupthaus auch kleine Flächen mit jeweils nur 4 Modulen. Ich will keine Optimierer und das Wagnis mit kurzen Strings werde ich mal zwischendurch testen, aber nicht als Dauerzustand lassen. Insofern löse ich diese kleine Flächen mit 3-phasigen Mikrowechselrichtern. Ich habe dazu auch endlich einen Elektriker gefunden, der eben String- und Mikrowechselrichter gemeinsam mit mir macht. Ist gar nicht so einfach da jemanden zu finden der nicht gleich die Augen verdreht ...

Da ich sicher noch weitere freie kleine Flächen erschließen werde, bitte unbedingt Deine Erfahrungen hier mitteilen. Mein Plan wäre eigentlich gewesen, dass mein PI die Daten von den OpenDTU's holt und der PI dann die Smartmeterdaten konsolidiert und phasenkorrekt an den Fronius schickt.

Solange Du aber nur eine OpenDTU hast, kann das natürlich auch über die OpenDTU direkt gemacht werden.

Viel Glück!

LG Michael

**Reply by Nulleinspeiser-mit-Batteriespeicher** [reply]
Hallo Abraham,

das ganze funktioniert ohne Probleme und ohne Peaks mit der letzten Version von https://github.com/StoRmtec500/OpenDTU 

Viele Grüße

Raphael

**Comment by Chilled** [comment]
Am Besten geht ihr gleich auf diesen Fork hier https://github.com/StoRmtec500/OpenDTU . Cyber hat aktuell keine Zeit für die notwendigen Anpassungen an die neuste OpenDTU Version. Stormi führt das aktuell fort

**Comment by Nulleinspeiser-mit-Batteriespeicher** [comment]
Hallo Chilled,

Das hatte ich in 

https://www.photovoltaikforum.com/thread/205384-hoymiles-mikrowechselrichter-mit-opendtu-in-solarweb-aufnehmen-ohne-smartmeter/

Auch gesehen, allerdings war zu dem Zeitpunkt das Peak-Problem noch nicht behoben. Ich danke dir für den Tipp, ich werde das weiter beobachten ob ich Erfahrungen zu dem Fork finden kann.

Viele Grüße 

Raphael

**Comment by Gitzo64** [comment]
Hallo Liebe Community,
ich weiß nicht ob ich hier richtig bin, aber ich versuche es einfach mal .
Ich habe eine PV Anlage mit 7 Hoymiles HMT 2250 aufgebaut, die
ich mit einer Open DTU abfrage.
Jetzt möchte ich die Daten gerne in meiner Siemens ET200SP CPU 1510 
per Modbus  TCP abfragen.
Hat die Open DTU Modbus TCP das in irgendeiner Version, derzeitig habe 
ich die neuste Version von der Open DTU on Battery installiert.
In der Version finde ich diese Schnittstelle leider nicht.
Bin über jede Hilfe dankbar.
MfG
Gitzo

**Reply by Chilled** [reply]
Die Version von Cyber ist für Fronius WR optimiert (Sunspec) Ob deine Siemens das kann musst du testen... [https://github.com/StoRmtec500/OpenDTU ](https://github.com/StoRmtec500/OpenDTU )
Hier noch der Forumsbeitrag [https://www.photovoltaikforum.com/thread/205384-hoymiles-mikrowechselrichter-mit-opendtu-in-solarweb-aufnehmen-ohne-smartmeter/](url)
